### DISCUSSION TITLE
Extract binary mask of crosswalks

### DISCUSSION THREAD

**Discussion by magicaltoast** [bdd100k/bdd100k#195] [Q&A] [answered]
I want to get a binary mask of crosswalks (1 where the pixel is a crosswalk, 0 when a pixel is not a crosswalk).
I don't care about other attributes of the lines, like direction and style.

I checked out [documentation of the format](https://doc.bdd100k.com/format.html#lane-mask).
But I don't really understand it because colormaps contain information in all 3 channels.

Any ideas on how I could achieve that?

**Comment by thomasehuang** [comment, marked as answer]
Hi. You can use the function provided [here](https://github.com/bdd100k/bdd100k/blob/master/bdd100k/eval/lane.py#L147) as an example. Something like `sub_task_funcs['category'](gt_mask, 0)` should return a binary mask for crosswalks. `gt_mask` is the ground truth mask, and `0` is the category id, which is 0 for crosswalks. For details on how it's implemented, you can see [this](https://github.com/bdd100k/bdd100k/blob/master/bdd100k/eval/lane.py#L119) function. Hope that helps!

**Reply by magicaltoast** [reply]
@thomasehuang Thanks, should I use masks or color masks? Because I can't figure out mask naming conventions, They don't match with image names.

**Reply by thomasehuang** [reply]
Masks. Colormaps are only for visualization purposes. The name of the mask should match with the image, besides the suffix `.jpg` vs `.png`.
